pkg/client/leaderelection/resourcelock/configmaplock.go

```diff
@@ -35,7 +35,7 @@ type ConfigMapLock struct {
 	// ConfigMapMeta should contain a Name and a Namespace of an
 	// ConfigMapMeta object that the Leadercmlector will attempt to lead.
 	ConfigMapMeta metav1.ObjectMeta
-	Client        clientset.Interface
+	Client        corev1client.ConfigMapsGetter
```

Comment:
@caesarxuchao, even though ConfigMapInterface is the same, as they are generated. they are 2 different types today as is evident from below error i got when i passed federation corev1 client.
```
# k8s.io/kubernetes/federation/cmd/federation-controller-manager/app
federation/cmd/federation-controller-manager/app/controllermanager.go:163: cannot use leaderElectionClient.CoreV1() (type "k8s.io/kubernetes/federation/client/clientset_generated/federation_clientset/typed/core/v1".CoreV1Interface) as type "k8s.io/kubernetes/pkg/client/clientset_generated/clientset/typed/core/v1".ConfigMapsGetter in field value:
	"k8s.io/kubernetes/federation/client/clientset_generated/federation_clientset/typed/core/v1".CoreV1Interface does not implement "k8s.io/kubernetes/pkg/client/clientset_generated/clientset/typed/core/v1".ConfigMapsGetter (wrong type for ConfigMaps method)
		have ConfigMaps(string) "k8s.io/kubernetes/federation/client/clientset_generated/federation_clientset/typed/core/v1".ConfigMapInterface
		want ConfigMaps(string) "k8s.io/kubernetes/pkg/client/clientset_generated/clientset/typed/core/v1".ConfigMapInterface
```
we should start using client-go, then these types will be unique